sdks/java/io/cdap/src/main/java/org/apache/beam/sdk/io/cdap/Plugin.java

```diff
@@ -84,16 +88,34 @@ public PluginConfig getPluginConfig() {
    * validating connection to the CDAP sink/source and performing initial tuning.
    */
   public void prepareRun() {
+    PluginConfig pluginConfig = getPluginConfig();
+    if (pluginConfig == null) {
```

Comment:
we should use preconditions.checkNotNull here as well

Reply:
Done